dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeLogsConnector.java

```diff
@@ -334,43 +334,48 @@ public void addTasksTo(List<? super Task<?>> out, ConnectorArguments arguments)
     out.add(new DumpMetadataTask(arguments, FORMAT_NAME));
     out.add(new FormatTask(FORMAT_NAME));
 
-    List<TaskDescription> tasks = new ArrayList<>();
-    if (arguments.isAssessment()) {
-      tasks.add(
-          new TaskDescription(
-              QueryHistoryExtendedFormat.ZIP_ENTRY_PREFIX,
-              createExtendedQueryFromAccountUsage(arguments),
-              QueryHistoryExtendedFormat.Header.class));
-    } else {
-      tasks.add(new TaskDescription(ZIP_ENTRY_PREFIX, newQueryFormat(arguments), Header.class));
-    }
-
-    if (arguments.isAssessment()) {
-      tasks.addAll(createTaskDescriptions(arguments));
-    }
-
     // (24 * 7) -> 7 trailing days == 168 hours
     // Actually, on Snowflake, 7 days ago starts at midnight in an unadvertised time zone. What the
     // <deleted>.
     // Snowflake will refuse (CURRENT_TIMESTAMP - 168 hours) because it is beyond the
     // 7-day window allowed by the server-side function.
     ZonedIntervalIterable intervals = ZonedIntervalIterable.forConnectorArguments(arguments);
     LOG.info("Exporting query log for " + intervals);
-    for (ZonedInterval interval : intervals) {
-      tasks.forEach(
-          task -> {
-            String query =
-                String.format(
-                    task.unformattedQuery,
-                    SQL_FORMAT.format(interval.getStart()),
-                    SQL_FORMAT.format(interval.getEndInclusive()));
-            String file =
-                task.zipPrefix
-                    + DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(interval.getStartUTC())
-                    + ".csv";
-            out.add(new JdbcSelectTask(file, query).withHeaderClass(task.headerClass));
-          });
+    for (ZonedInterval zonedInterval : intervals) {
+      TaskDescription taskDescription;
+      if (arguments.isAssessment()) {
+        taskDescription =
+            new TaskDescription(
+                QueryHistoryExtendedFormat.ZIP_ENTRY_PREFIX,
+                createExtendedQueryFromAccountUsage(arguments),
+                QueryHistoryExtendedFormat.Header.class);
+        addJdbcTask(out, zonedInterval, taskDescription);
```

Comment:
I believe this line is redundant?

Reply:
Sorry it seems I was reviewing an outdated version. Please disregard.

Reply:
no worries. github to blame